dcicutils/glacier_utils.py

```diff
@@ -0,0 +1,321 @@
+                        response = self.s3.delete_object(
+                            Bucket=bucket,
+                            Key=key,
+                            VersionId=v.get('VersionId')
+                        )
+                        PRINT(f'Object {key} Glacier version {v.get("VersionId")} deleted:\n{response}')
+                        break
+                else:
+                    PRINT(f'No Glacier version found for object {key}')
+
+        except Exception as e:
+            PRINT(f'Error deleting Glacier versions of object {key}: {str(e)}')
+            return False
+
+    def copy_object_back_to_original_location(self, bucket: str, key: str,
+                                              version_id: Union[str, None] = None) -> Union[dict, None]:
+        """ Reads the temporary location from the restored object and copies it back to the original location
+
+        :param bucket: bucket where object is stored
+        :param key: key within bucket where object is stored
+        :param version_id: version of object, if applicable
+        :return: boolean whether the copy was successful
+        """
+        try:
+            # Get temporary location
+            restore = self.extract_temporary_s3_location_from_restore_response(bucket, key)
+            # Copy the object from the temporary location to its original location
+            copy_source = {'Bucket': bucket, 'Key': key, 'Restore': restore}
+            copy_args = {'Bucket': bucket, 'Key': key}
+            if version_id:
+                copy_source['VersionId'] = version_id
+                copy_args['CopySourceVersionId'] = version_id
+            response = self.s3.copy_object(CopySource=copy_source, **copy_args)
+            PRINT(f'Response from boto3 copy:\n{response}')
+            PRINT(f'Object {bucket}/{key} copied back to its original location in S3')
+            return response
+        except Exception as e:
+            PRINT(f'Error copying object {bucket}/{key} back to its original location in S3: {str(e)}')
+            return None
+
+    def restore_glacier_phase_one_restore(self, atid_list: List[str], days: int = 7) -> (List[str], List[str]):
+        """ Triggers a restore operation for all @id in the @id list, returning a list of success and
+            error objects.
+
+        :param atid_list: list of @ids to restore from glacier
+        :param days: days to store the temporary copy
+        :return: 2 tuple of success, error list of @ids
+        """
+        success, errors = [], []
+        for atid in atid_list:
+            resp = self.restore_portal_from_glacier(atid, days=days)
+            if resp:
+                success.append(atid)
+            else:
+                errors.append(atid)
+        if len(errors) == 0:
+            PRINT(f'Errors encountered restoring @ids: {errors}')
+        else:
+            PRINT(f'Successfully triggered restore requests for all @ids passed {success}')
+        return success, errors
```

Comment:
This kind of passive result strategy is not my personal preference. I'd rather just see success communicated as normal return and errors by raising. If the issue is raising multiple errors, see the `MultiError` class that I wrote in my [PR #250](https://github.com/4dn-dcic/utils/pull/250).

Reply:
My thinking is the caller should handle errors as needed? I want something that won't crash mid run. Having this go part way can be dangerous - special care is needed when using this.